src/tools/compiletest/src/header.rs

```diff
@@ -642,14 +644,42 @@ impl Config {
     /// Parses a name-value directive which contains config-specific information, e.g., `ignore-x86`
     /// or `normalize-stderr-32bit`.
     fn parse_cfg_name_directive(&self, line: &str, prefix: &str) -> ParsedNameDirective {
+        // This matches optional whitespace, followed by a group containing a series of word
+        // characters (including '_' and '-'), followed optionally by a sequence consisting
+        // of a colon, optional whitespace, and another group containing word characters.
+        //
+        // Matches in full:
+        //   cfg-target-has-atomic: 128
+        //   cfg-target-has-atomic:128
+        //   cfg-target-has-atomic
+        //
+        // Matches up to the first space (exclusive):
+        //   ignore-test - This test really shouldn't ever be run
+        //
+        // Matches up to the second space (exclusive):
+        //   cfg-target-has-atomic: 128 - this requires fancy newfangled atomics
+        //
+        // Matches up to the second colon (exclusive)
```

Comment:
I'm not following the intent of this case - what is this useful for?

Reply:
Comments. There are a bunch of situations where we're using comments on the same line as a directive, using `- comment` or whatever. I didn't want to break those. I could leave a comment here explaining that?

Reply:
Those aren't present on any cfg- lines though, right?

I don't think we should use a colon to terminate lines anyway, that seems "weird". Ultimately I'm not too worried about supporting comments *on the same line* -- the only worry would be if the commit causes us to silently not run a test (which the author may not notice). Not sure how much we can detect mistakes here though... I recall there was some logic added to this effect in Cargo(?) recently, but not 100% sure if that works for us or we can use it...